Mathlib/RingTheory/HahnSeries/Order.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2025 Weiyi Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weiyi Wang
+-/
+
+import Mathlib.Algebra.Order.Group.Abs
+import Mathlib.Algebra.Order.Monoid.Defs
+import Mathlib.Order.PiLex
+import Mathlib.RingTheory.HahnSeries.Addition
+
+/-!
+
+# Lexicographical order on Hahn series
+
+In this file, we equip `HahnSeries Γ R` with lexicographical order, and show this
+is a `LinearOrder` when `Γ` and `R` themselves are linearly ordered. Additionally,
+it is an ordered group when `R` is.
+
+-/
+
+namespace HahnSeries
+
+section PartialOrder
+
+variable {Γ : Type*} {R : Type*}
+variable [LinearOrder Γ] [Zero R] [PartialOrder R]
+
+variable (Γ R) in
+instance instPartialOrder : PartialOrder (HahnSeries Γ R) :=
+  PartialOrder.lift (toLex ·.coeff) (by
+    intro x y
+    simp
+  )
+
+theorem lt_iff (a b : HahnSeries Γ R) :
+    a < b ↔ ∃ (i : Γ), (∀ (j : Γ), j < i → a.coeff j = b.coeff j) ∧ a.coeff i < b.coeff i := by rfl
+
+end PartialOrder
+
+section LinearOrder
+
+variable {Γ : Type*} {R : Type*}
+variable [LinearOrder Γ] [Zero R] [LinearOrder R]
+
+variable (Γ R) in
+noncomputable
+instance instLinearOrder : LinearOrder (HahnSeries Γ R) where
```

Comment:
My initial instinct is that this should be
```suggestion
instance instLinearOrder : LinearOrder (Lex (HahnSeries Γ R)) where
```
which is certainly more conservative. Of course, this will make the rest of your development a little bit more annoying with extra `toLex` and `ofLex`es.

Do we put global orders on `PowerSeries`, `LaurentSeries`, etc in mathlib? We should probably be consistent here.

Reply:
My recommendation would be to just do that, and perhaps discuss removing it later if it's really annoying. That discussion will be easier to have once there are applications to point at.